addons/barcodes/models/barcodes.py

```diff
@@ -186,26 +178,18 @@ def _encoding_selection_list(self):
                 ('upca', 'UPC-A'),
         ]
 
-    @api.model
     def _get_type_selection(self):
         return [('alias', _('Alias')), ('product', _('Unit Product'))]
 
-    _columns = {
-        'name':     fields.char('Rule Name', size=32, required=True, help='An internal identification for this barcode nomenclature rule'),
-        'barcode_nomenclature_id':     fields.many2one('barcode.nomenclature','Barcode Nomenclature'),
-        'sequence': fields.integer('Sequence', help='Used to order rules such that rules with a smaller sequence match first'),
-        'encoding': fields.selection('_encoding_selection_list','Encoding',required=True,help='This rule will apply only if the barcode is encoded with the specified encoding'),
-        'type':     fields.selection('_get_type_selection','Type', required=True),
-        'pattern':  fields.char('Barcode Pattern', size=32, help="The barcode matching pattern", required=True),
-        'alias':    fields.char('Alias',size=32,help='The matched pattern will alias to this barcode', required=True),      
-    }
-
-    _defaults = {
-        'type': 'product',
-        'pattern': '.*',
-        'encoding': 'any',
-        'alias': "0",
-    }
+    _type_selection = lambda self: self._get_type_selection()
+
+    name = fields.Char(string='Rule Name', size=32, required=True, help='An internal identification for this barcode nomenclature rule')
+    barcode_nomenclature_id = fields.Many2one('barcode.nomenclature', string='Barcode Nomenclature')
+    sequence = fields.Integer(string='Sequence', help='Used to order rules such that rules with a smaller sequence match first')
+    encoding = fields.Selection(_encoding_selection_list, string='Encoding', required=True, default='any', help='This rule will apply only if the barcode is encoded with the specified encoding')
```

Comment:
this change is incorrect, the method should still be passed as a name otherwise it can't be overridden. However the `@api.multi` on the method can be removed.